nbextensions/appCell2/widgets/appCellWidget.js

```diff
@@ -2015,9 +2018,10 @@ define(
                          */
                         const jobState = model.getItem('exec.jobState');
                         if (jobState && !Jobs.isValidJobStateObject(jobState)) {
-                            // use the 'created' key to see if it's an updated jobState
                             startListeningForJobMessages(jobState.job_id);
                             requestJobStatus(jobState.job_id);
+                        } else if (model.getItem('exec.launchState.job_id')) {
+                            startListeningForJobMessages(jobState.job_id);
```

Comment:
I have seen an error where the app cell is in the `internal-error` state, with nothing in `exec.jobState` but a job ID in `exec.launchState.job_id`. This may enable the cell to recover from that state (if the FSM allows it, of course).